src/net.cpp

```diff
@@ -1573,31 +1579,61 @@ void CConnman::ThreadDNSAddressSeed()
     if (gArgs.GetBoolArg("-forcednsseed", DEFAULT_FORCEDNSSEED)) {
         // When -forcednsseed is provided, query all.
         seeds_right_now = seeds.size();
+    } else if (addrman.size() == 0) {
+        // If we have no known peers, query all.
+        seeds_right_now = seeds.size();
```

Comment:
In #15558 the first run will set `seeds_right_now = 0` then see that `addrman.size() > 0 && seeds_right_now == 0` is false (since `addrman.size() == 0`) so will not sleep nor bump `seeds_right_now` by `DNSSEEDS_TO_QUERY_AT_ONCE`, it will then query a seed, and decrement `seeds_right_now` so that it's `-1`, at which point `addrman.size() > 0 && seeds_right_now == 0` will continue being false (because `seeds_right_now < 0`) and will query all seeds without waiting between them.

This retains that behaviour, but is explicit about it. See [suhas's comment on 15558](https://github.com/bitcoin/bitcoin/pull/15558#discussion_r327643907)